src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -6970,6 +6972,34 @@ instruct cmovI_regUCF(cmpOpUCF cop, rFlagsRegUCF cr, rRegI dst, rRegI src) %{
   %}
 %}
 
+instruct cmovI_regUCF2_ne(cmpOpUCF2 cop, rFlagsRegUCF cr, rRegI dst, rRegI src) %{
+  predicate(n->in(1)->in(1)->as_Bool()->_test._test == BoolTest::ne);
+  match(Set dst (CMoveI (Binary cop cr) (Binary dst src)));
+
+  ins_cost(200); // XXX
+  format %{ "cmovpl  $dst, $src\n\t"
+            "cmovnel $dst, $src" %}
+  ins_encode %{
+    __ cmovl(Assembler::parity, $dst$$Register, $src$$Register);
+    __ cmovl(Assembler::notEqual, $dst$$Register, $src$$Register);
+  %}
+  ins_pipe(pipe_cmov_reg);
+%}
+
+instruct cmovI_regUCF2_eq(cmpOpUCF2 cop, rFlagsRegUCF cr, rRegI dst, rRegI src) %{
+  predicate(n->in(1)->in(1)->as_Bool()->_test._test == BoolTest::eq);
+  match(Set dst (CMoveI (Binary cop cr) (Binary src dst)));
+
+  ins_cost(200); // XXX
+  format %{ "cmovpl  $dst, $src\n\t"
+            "cmovnel $dst, $src" %}
+  ins_encode %{
+    __ cmovl(Assembler::parity, $dst$$Register, $src$$Register);
+    __ cmovl(Assembler::notEqual, $dst$$Register, $src$$Register);
```

Comment:
Should this be `equal`?

Reply:
I see that you swapped `src, dst` in `match()` but `format` is sill incorrect and the code is confusing.

Reply:
This is a flipping the sense of the test by flipping the input of the `CMove`, so this is essentially the same as in the above rule. Thanks.

Reply:
I agree with @vnkozlov that this needs explanation. Could you please add comments here with IR and example code generated for both the eq case and ne case? You have some explanation in the PR description but not in the code. The description needs to be in the code as well for maintenance.

Reply:
Right, I missed that. Then you can use `expand %{` to avoid duplication (keep format).